packages/core/src/components/graphs/gauge.ts

```diff
@@ -0,0 +1,288 @@
+			return 0;
+		}
+		return -arcSize / 2;
+	}
+
+	render(animate = true) {
+		const self = this;
+		const svg = this.getContainerSVG();
+		const options = this.model.getOptions();
+		const ratio = this.getCurrentRatio();
+		const datalist = this.getDataList();
+		const delta = this.getDelta();
+		const arcSize = this.getArcSize();
+		const startAngle = this.getStartAngle();
+		const rotationAngle = ratio * arcSize;
+		const currentAngle = startAngle + rotationAngle;
+		const endAngle = startAngle + arcSize;
+
+		// Compute the outer radius needed
+		const radius = this.computeRadius();
+
+		this.backgroundArc = arc()
+			.innerRadius(this.getInnerRadius())
+			.outerRadius(radius)
+			.startAngle(currentAngle)
+			.endAngle(endAngle);
+
+		this.arc = arc()
+			.innerRadius(this.getInnerRadius())
+			.outerRadius(radius)
+			.startAngle(startAngle)
+			.endAngle(currentAngle);
+
+		// Set the hover arc radius
+		this.hoverArc = arc()
+			.innerRadius(this.getInnerRadius())
+			.outerRadius(radius + options.pie.hoverArc.outerRadiusOffset)
+			.startAngle(startAngle)
+			.endAngle(currentAngle);
+
+		// Add background arc
+
+		DOMUtils.appendOrSelect(svg, "g.background")
+			.append("path")
+			.attr("d", this.backgroundArc)
+			.attr("fill", "#bbb")
+			.attr("role", Roles.GROUP);
+
+		// Add data arc
+
+		DOMUtils.appendOrSelect(svg, "g.arc")
+			.append("path")
+			.data(datalist)
+			.attr("d", this.arc)
+			.classed("arc", true)
+			.attr("fill", "#00f");
+
+		// Position Pie
+		const gaugeTranslateX = radius + options.pie.xOffset;
+		const gaugeTranslateY = radius + options.pie.yOffset;
+		svg.attr("transform", `translate(${gaugeTranslateX}, ${gaugeTranslateY})`);
+
+		// Add the number shown in the center of the gauge and the delta
+		// under it.
+
+		DOMUtils.appendOrSelect(svg, "text.gauge-value")
+			.attr("text-anchor", "middle")
+			.attr("alignment-baseline", "middle")
+			.style("font-size", () => options.gauge.center.valueFontSize(radius, arcSize))
+			.transition(this.services.transitions.getTransition("donut-figure-enter-update", animate))
+			.attr("y", options.gauge.center.valueYPosition(radius, arcSize))
+			.tween("text", function() {
+				return self.centerNumberTween(select(this), ratio * 100);
+			});
+
+		DOMUtils.appendOrSelect(svg, "text.gauge-delta")
+			.attr("text-anchor", "middle")
+			.attr("alignment-baseline", "middle")
+			.style("font-size", () => options.gauge.center.deltaFontSize(radius, arcSize))
+			.attr("y", options.gauge.center.deltaYPosition(radius, arcSize))
+			.text(() => {
+				const sign = Math.sign(delta)
+				const deltaPerc = (delta * 100).toFixed(2);
+				const arrow = sign === 0 ? "=" : sign === -1 ? "▼" : "▲";
+				return `${arrow}  ${deltaPerc}%`;
+			});
+
+		// Add event listeners
+		this.addEventListeners();
+	}
+
+
+	getInnerRadius() {
+		// Compute the outer radius needed
+		const radius = this.computeRadius();
+		return radius * (3 / 4);
+	}
+
+	centerNumberTween(d3Ref, value: number) {
```

Comment:
looks a bit weird to me passing `d3Ref` here, maybe rearrange this func in order to pass and return just the value?

Reply:
it looked weird to me too, but I simply kept the same function from the donut chart, if there is no problem in changing it I'll do